### DISCUSSION TITLE
how to know a client connects to/ disconnects to as a gatt server?

### DISCUSSION THREAD

**Discussion by YFJiang** [bluez/bluer#150] [Q&A] [answered]
Hello,
how to know a client connects to/ disconnects to as a gatt server? Is there a session exists inside BlueR?

**Comment by surban** [comment, marked as answer]
There is no way to know whether a client is connected to a GATT server in BlueZ.

If you need that information, the client should subscribe to a characteristic, which will cause a notification session to start. See [CharacteristicNotify](https://docs.rs/bluer/latest/bluer/gatt/local/struct.CharacteristicNotify.html).

**Reply by YFJiang** [reply]
Thanks. But no way to know when a client is disconnected, right?

**Reply by surban** [reply]
If you use the IO-based method, you can use [`CharacteristicWriter::closed`](https://docs.rs/bluer/latest/bluer/gatt/struct.CharacteristicWriter.html#method.closed) to be notified when the notification session ends.
